devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy

```diff
@@ -286,6 +284,33 @@ private int get_LOG_TRACE() {
 	return 5
 }
 
+private def convertVoltToPercent(value) {
+	def levelValue;
+	def levelsTable = [0, 20000, 40000, 60000, 80000, 100000];
+	def anglesTable = [30, 55, 56, 57, 58.5, 60];
+
+	if (value > anglesTable[anglesTable.size - 1]) { // if the value of the angle is greater than the maximum
+	  value = anglesTable[anglesTable.size - 1]; // use the maximum value instead
+	}
+
+	def index = 1;
```

Comment:
Shouldn't the index start at 0 ?

Reply:
Like I said a week or 2 ago, Vincent is not working for us anymore. I don't have note for this DTH so I don't know why he used 1 for the index, but after taking a quick look at the code, a couple of lines below, you can see he use [index -1] so if we put it at 0 at first we will end up with a negative index at some point.

Reply:
thanks, I send a notification to @vbeauregard-Sinope because I do not know who from Sinope I can send a reminder